app/src/main/java/com/io1/todo/feature/onboarding/view/OnboardingScreen.kt

```diff
@@ -1,42 +1,121 @@
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxHeight
 import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.MaterialTheme
 import androidx.compose.material3.Button
-import androidx.compose.material3.Scaffold
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.google.accompanist.pager.ExperimentalPagerApi
+import com.google.accompanist.pager.HorizontalPager
+import com.google.accompanist.pager.HorizontalPagerIndicator
+import com.io1.todo.feature.onboarding.viewmodel.OnboardingViewModel
+import com.io1.todo.feature.onboarding.utils.OnBoardingPage
 
+@ExperimentalAnimationApi
+@ExperimentalPagerApi
 @Composable
 fun OnboardingScreenRoute(onMainButtonClick: () -> Unit) {
     OnboardingScreen(onMainButtonClick = onMainButtonClick)
 }
 
 @SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
+@ExperimentalAnimationApi
+@ExperimentalPagerApi
 @Composable
-fun OnboardingScreen(onMainButtonClick: () -> Unit = {}) {
-    Scaffold(modifier = Modifier
-        .fillMaxSize(),
-        containerColor = Color.Black,
-        topBar = {
+fun OnboardingScreen(
+    onMainButtonClick: () -> Unit = {},
+    onboardingViewModel: OnboardingViewModel = hiltViewModel()
+) {
+    val pages = listOf(
+        OnBoardingPage.First,
+        OnBoardingPage.Second,
+        OnBoardingPage.Third
+    )
+
+    val pagerState = com.google.accompanist.pager.rememberPagerState()
 
-        },
-        content = {
-            Column(
-                modifier = Modifier.fillMaxSize(),
-                verticalArrangement = Arrangement.Center,
-                horizontalAlignment = Alignment.CenterHorizontally
-            ) {
-                Text(text = "Onboarding", color = Color.White)
-                Button(onClick = { onMainButtonClick() }) {
+    Column (modifier = Modifier.fillMaxSize()){
+        HorizontalPager(
+            modifier = Modifier.weight(10f),
+            count = 3,
+            state = pagerState,
+            verticalAlignment = Alignment.Top
+        ) { position ->
+            PagerScreen(onBoardingPage = pages[position])
+        }
+        HorizontalPagerIndicator(
+            modifier = Modifier
+                .align(Alignment.CenterHorizontally)
+                .weight(1f),
+            pagerState = pagerState
+        )
+        Row(
+            modifier = Modifier
+                .padding(horizontal = 40.dp),
```

Comment:
I would suggest that all dimensions or sp values should come from themes or resources